File: frontend/lib/graphql/queries/document.dart
const String getDocuments = r'''
query getDocuments($input: GetDocumentsInput!){
    getDocuments(input:$input){
        h_id, 
        type{
            id,
            name_en,
            name_ro
        }         
        series,
        number,
        date,
        due_date,
        notes,
        deleted,
        efactura_status,
        partner{
            id,
            code,        
            type,        
            active,
            name,
            vat,
            vat_number,
            registration_number,
            individual_number 
        }      
    }
}
''';

const String getDocumentById = r'''
query  getDocumentById($documentId: String!) {
   getDocumentById(documentId: $documentId){
    h_id,
    type{
            id,
            name_ro,
            name_en
        } 
    series,
    number,
    date,
    notes,
    deleted,
    efactura_status,
    partner{
        id,
        code,        
        type,        
        active,
        name,
        vat,
        vat_number,
        registration_number,
        individual_number 
    },
    document_items{
        d_id,
        item{           
            id,
            code,
            name,
            is_active,
            is_stock,
            um{
                id,
                name,
                code,
                is_active
             },
            vat{
                id,
                name,
                percent
            },
        }
      
      quantity,
      price,     
      amount_net,
      amount_vat,
      amount_gross,
      item_type_pn
    }  
   }
}
''';

const String getDocumentTransactions = r'''
query getDocumentTransactions{
    getDocumentTransactions{
        id,
        name,
        document_type_source_id,
        document_type_destination_id
    }
}
''';

const String getGenerateAvailableItems = r'''
query  getGenerateAvailableItems($input: GetGenerateAvailableItemsInput!) {
   getGenerateAvailableItems(input:$input){
    h_id,  
    series,
    number,
    date,
    document_item{
        d_id,
        item{
          id,
          code,
          name,
          um{
            id,
            name
          }
          vat{
            id,
            name,
            percent
          }
        }     
        quantity,      
    }  
  }
}
''';

File: backend/models/document.go
package models

type Document struct {
	HId            string       `json:"h_id"`
	Type           DocumentType `json:"type"`
	Series         *string      `json:"series"`
	Number         string       `json:"number"`
	Date           string       `json:"date"`
	DueDate        *string      `json:"due_date"`
	Deleted        bool         `json:"deleted"`
	EFacturaStatus *string      `json:"efactura_status"`
	Notes          *string      `json:"notes"`
	//DocumentItems []*DocumentItem `json:"document_items"`
	//Person           *Company    `json:"company"`
	//Individual         *Individual `json:"company"`
}

type DocumentItem struct {
	DId         string   `json:"d_id"`
	Item        Item     `json:"item"`
	Quantity    float64  `json:"quantity"`
	Price       *float64 `json:"price"`
	AmountNet   *float64 `json:"amount_net"`
	AmountVat   *float64 `json:"amount_vat"`
	AmountGross *float64 `json:"amount_gross"`
	ItemTypePn  *string  `json:"item_type_pn,omitempty"`
}

type DocumentType struct {
	ID     int    `json:"id"`
	NameRo string `json:"name_ro"`
	NameEn string `json:"name_en"`
}
